tm2/pkg/bft/rpc/lib/server/handlers.go

```diff
@@ -191,9 +195,16 @@ func makeJSONRPCHandler(funcMap map[string]*RPCFunc, logger *slog.Logger) http.H
 			}
 			responses = append(responses, types.NewRPCSuccessResponse(request.ID, result))
 		}
-		if len(responses) > 0 {
+		if len(responses) == 0 {
+			return
+		}
+
+		if isRPCRequests {
 			WriteRPCResponseArrayHTTP(w, responses)
+			return
 		}
+
+		WriteRPCResponseHTTP(w, responses[0])
```

Comment:
This code looks like it could be replaced by something like:
```go
if len(responses) == 1 {
  WriteRPCResponseHTTP(w, responses[0])
} else if if len(responses) > 1 {
  WriteRPCResponseArrayHTTP(w, responses)
}
```

Or maybe it would better to check this directly inside `WriteRPCResponseArrayHTTP` function and to just let the code as it is in this file.

Reply:
I just checked the `WriteRPCResponseArrayHTTP ` and it already check this https://github.com/gnolang/gno/blob/d75f1a2fe060c64f0ed51182f5dda965043ace9f/tm2/pkg/bft/rpc/lib/server/http_server.go#L122-L124

I'm not sure that these changes in the code don't have any impact then.

Reply:
:o I forget to commit this https://github.com/gnolang/gno/pull/3678/commits/7453add45b8d0de073bfc3cbcfa619f2ca172206 is why it failing   sorry ^^

Reply:
I haven't tried re-running the script on this PR because I really feel like this commit won't fix it.
Did you try it on your side? Is it working?

Reply:
Edited the test + add a subtest here: https://github.com/gnolang/gno/pull/3678/commits/b37dc8cae55f13650090aa3307a2e4e963e32d6d

Reply:
> I haven't tried re-running the script with this change because I really feel like this commit won't fix it. Did you try it on your side? Is it working?

Yes it works

<img width="1224" alt="image" src="https://github.com/user-attachments/assets/f7e0f0a6-0d9c-4e94-9898-40b9df540c2c" />

<img width="1257" alt="image" src="https://github.com/user-attachments/assets/b2fb98d1-4c11-4324-a61b-2e8673c168af" />

Reply:
Ok cool :) I'll check it

Reply:
Ok, it works well, GG 👍 

So your fix is not about the number of responses you have to send but whether you received an array or not. If you get an array of size 1, you should return an array of size 1, right?

If that's the case, could you change your boolean name to something like `isRPCRequestArray` and add a comment right above your last if explaining that for a request in the form of an array, even if it contains only one element, your response must also be an array.

Thanks!

Reply:
> So your fix is not about the number of responses you have to send but whether you received an array or not. If you get an array of size 1, you should return an array of size 1, right?


Yes 

> If that's the case, could you change your boolean name to something like isRPCRequestArray and add a comment right above your last if explaining that for a request in the form of an array, even if it contains only one element, your response must also be an array.


I used "isRPCRequests" with an added "s", but "isRPCRequestArray" is clearer.
 I’ve added a comment here: https://github.com/gnolang/gno/pull/3678/commits/50fa2cdbbfe73746bfe2a8d08da694593fa1aec4.